addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -678,12 +668,12 @@ def _action_validate_create_childs(self):
                 mail_activity_automation_skip=True
             ).create(allocation_create_vals)
             if childs:
-                childs.action_validate()
+                childs.filtered(lambda child: child.state == 'confirm').action_validate()
```

Comment:
Some allocation does not require any manual validation to be validated (depending on the `validation_type` field). If the allocations are not automatically. Since the method is supposed to be only used on one record at a time, I changed the condition instead to only validate the children on allocation that require it.